Dolphin scripts/Entrance Randomizer/lib/entrance_rando.py

```diff
@@ -40,6 +47,131 @@ class Transition(NamedTuple):
+    # because we don't want to fight any given animal spirit more than once in a run
+    (LevelCRC.MONKEY_TEMPLE, LevelCRC.MONKEY_SPIRIT),
+    (LevelCRC.MONKEY_SPIRIT, LevelCRC.MONKEY_TEMPLE),
+    (LevelCRC.SCORPION_TEMPLE, LevelCRC.SCORPION_SPIRIT),
+    (LevelCRC.SCORPION_SPIRIT, LevelCRC.SCORPION_TEMPLE),
+    (LevelCRC.PENGUIN_TEMPLE, LevelCRC.PENGUIN_SPIRIT),
+    (LevelCRC.PENGUIN_SPIRIT, LevelCRC.PENGUIN_TEMPLE),
+    # The 5 Native Games are currently chosen to not be randomized.
+    # If we at some point decide to randomize them anyway we'll have to do some rigorous testing
+    # Because it's very much possible this will cause some bugs
+    (LevelCRC.NATIVE_VILLAGE, LevelCRC.WHACK_A_TUCO),
+    (LevelCRC.WHACK_A_TUCO, LevelCRC.NATIVE_VILLAGE),
+    (LevelCRC.NATIVE_VILLAGE, LevelCRC.TUCO_SHOOT),
+    (LevelCRC.TUCO_SHOOT, LevelCRC.NATIVE_VILLAGE),
+    (LevelCRC.NATIVE_VILLAGE, LevelCRC.RAFT_BOWLING),
+    (LevelCRC.RAFT_BOWLING, LevelCRC.NATIVE_VILLAGE),
+    (LevelCRC.NATIVE_VILLAGE, LevelCRC.PICKAXE_RACE),
+    (LevelCRC.PICKAXE_RACE, LevelCRC.NATIVE_VILLAGE),
+    (LevelCRC.NATIVE_VILLAGE, LevelCRC.KABOOM),
+    (LevelCRC.KABOOM, LevelCRC.NATIVE_VILLAGE),
+    # The 2 CUTSCENE Levels are currently chosen to not be randomized.
+    # As of right now both of these cutscenes are hijacked to be skipped entirely
+    (LevelCRC.JAGUAR, LevelCRC.PLANE_CUTSCENE),
+    (LevelCRC.PLANE_CUTSCENE, LevelCRC.CRASH_SITE),
+    (LevelCRC.SPINJA_LAIR, LevelCRC.VIRACOCHA_MONOLITHS_CUTSCENE),
+    (LevelCRC.VIRACOCHA_MONOLITHS_CUTSCENE, LevelCRC.VIRACOCHA_MONOLITHS),
+    # Specific one-time, one-way warps are not randomized.
+    # These 3 are handled elsewhere in the randomizer
+    # Currently there are no plans to randomize these transitions.
+    (LevelCRC.ALTAR_OF_AGES, LevelCRC.BITTENBINDERS_CAMP),
+    (LevelCRC.ST_CLAIRE_DAY, LevelCRC.ST_CLAIRE_NIGHT),
+    (LevelCRC.ST_CLAIRE_NIGHT, LevelCRC.ST_CLAIRE_DAY),
+    # The Endgame El Dorado transitions are not randomized.
+    # Currently there are no plans to randomize these transitions.
+    (LevelCRC.GATES_OF_EL_DORADO, LevelCRC.JAGUAR),
+    (LevelCRC.JAGUAR, LevelCRC.PUSCA),
+    (LevelCRC.PUSCA, LevelCRC.GATES_OF_EL_DORADO),
+    # The Unused Beta Volcano Level is not randomized yet,
+    # but this can absolutely be randomized later at some point.
+    # It will require some special code though.
+    (LevelCRC.BETA_VOLCANO, LevelCRC.JUNGLE_CANYON),
+    (LevelCRC.BETA_VOLCANO, LevelCRC.PLANE_COCKPIT),
+]
+
+TRANSITION_INFOS_DICT_RANDO = TRANSITION_INFOS_DICT.copy()
+ALL_POSSIBLE_TRANSITIONS_RANDO = ALL_POSSIBLE_TRANSITIONS
+
+
+def remove_disabled_exits():
+    # remove exits from TRANSITION_INFOS_DICT_RANDO
+    for area in TRANSITION_INFOS_DICT.values():
+        for ex in area.exits:
+            current = (area.area_id, ex.area_id)
+            if current in one_way_exits or current in disabled_exits:
+                TRANSITION_INFOS_DICT_RANDO[area.area_id].exits = (
```

Comment:
There's no "tuple comprehension", so this actually creates a generator

```suggestion
                TRANSITION_INFOS_DICT_RANDO[area.area_id].exits = tuple(
```